source/elements/oneMKL/source/domains/sparse_linear_algebra.inc.rst

```diff
@@ -9,29 +9,23 @@ Sparse Linear Algebra
 
 .. container::
 
-
-   The oneAPI Math Kernel Library provides a Data Parallel C++ interface
-   to some of the Sparse Linear Algebra routines.
-
-   :ref:`onemkl_spblas` provides basic operations on sparse vectors and matrices, and
-   separates them into two stages: analysis
-   (also called inspector stage or optimize stage) and execution. For a given matrix,
-   the analysis would typically be called one time and the execution may be called
-   multiple times. During the analysis stage, the API inspects the matrix properties
-   including size, sparsity pattern and available parallelism and can apply matrix
-   format or structure changes to enable a more optimized algorithm.
-   In the execution stage, multiple routine calls can take advantage of the analysis
-   stage data in order to improve performance.
-
-
-   In order to save information in between calls to Sparse BLAS computation routines,
-   the :ref:`onemkl_sparse_matrix_handle_t` type is introduced, that is essentially
-   an opaque pointer, used to store data related to initial sparse matrix
-   and data obtained during analysis stage.
-
+   The oneAPI Math Kernel Library provides a C++ interface to some Sparse Linear
+   Algebra routines using SYCL.
+
+   :ref:`onemkl_spblas` provides basic operations on sparse vectors and
+   matrices, and separates them into three stages: query of the external
+   workspace size, optimization stage and execution. For a given configuration,
+   the first two stages would typically be called once for a set of input
```

Comment:
I know we currently do not have matmat (sparse * sparse -> sparse) listed in the API set, but it adds some complexity that would contradict the above statements about first two stages.  It might be useful to have a discussion about what matmat or maybe we will call it spgemm might look like in this new API.  The operation will be "C = alpha * op(A) * op(B) + beta * D"   where C and D may be different or the same handle, but if C and D are the same and beta is not zero, then op(A) * op(B) pattern must be a subset of C's pattern. 

The tricky part here is that we have essentially three stages -- work_estimation, compute and finalize.  Consider if beta is zero or C != D, then the size of C is unknown at the beginning and must be computed.  The computation is expensive and involves several kernels and full pass through the data just to figure out an upper bound on the sizes of output.  At that point, we can request a user provided workspace that is dependent on that computed size...  We can then use that workspace to fill in the computation in parallel and give user correct sizes to allocate their C matrix arrays and provide them to us to fill...  it is a rather complicated operation but is similar for sparse matrix addition, since again the final size is not known without some computations traversing the matrix data structures themselves...

Reply:
I have made this description just a bit more vague to specify that most operations are split into 3 stages in https://github.com/oneapi-src/oneAPI-spec/pull/522/commits/4b66461cc71418bedd4c0f0dc6a99b596b27fd72. I think it is useful to give a general overview here.
I think what you are describing can be added once we are ready to support matmat.